pkg/kubectl/cmd/set/set_image.go

```diff
@@ -164,12 +166,21 @@ func (o *ImageOptions) Run() error {
 		transformed := false
 		_, err := o.UpdatePodSpecForObject(info.Object, func(spec *api.PodSpec) error {
 			for name, image := range o.ContainerImages {
-				containerFound := false
+				var (
+					containerFound bool
+					err            error
+					resolved       string
+				)
 				// Find the container to update, and update its image
 				for i, c := range spec.Containers {
-					if c.Name == name || name == "*" {
-						spec.Containers[i].Image = image
-						containerFound = true
+					if containerFound = c.Name == name || name == "*"; containerFound {
+						if len(resolved) == 0 {
+							if resolved, err = o.ResolveImage(image); err != nil {
+								allErrs = append(allErrs, fmt.Errorf("error: unable to set image for container %q: %v", name, err))
```

Comment:
fixed (although i don't like we print the image name as the input might not really be an image but whatever the resolver deal with (image stream tag, image stream image, ???)